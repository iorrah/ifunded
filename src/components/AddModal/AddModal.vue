<template>
  <div class="modal__overlay">
    <div class="modal__aligner">
      <div class="modal__card">
        <div class="modal__header">
          <h2 class="modal__title">Add user</h2>

          <img
            src="../../assets/cross.svg"
            class="modal__close-modal"
          />
        </div>

        <div class="modal__body">
          <div class="material-form">
            <div class="material-form__input-text modal__input-text--first-name">
              <div class="group">
                <input
                  data-message="The format of your email is incorrect"
                  type="text"
                  placeholder=""
                  required=""
                  class="inpField"
                  v-model="firstName"
                />

                <span class="highlight"></span>
                <span class="bar"></span>

                <label>
                  First name
                </label>

                <span class="error-message">
                  Error
                </span>
              </div>
            </div>

            <div class="material-form__input-text modal__input-text--last-name">
              <div class="group">
                <input
                  data-message="The format of your email is incorrect"
                  type="text"
                  placeholder=""
                  required=""
                  class="inpField"
                  v-model="lastName"
                />

                <span class="highlight"></span>
                <span class="bar"></span>

                <label>
                  Last name
                </label>

                <span class="error-message">
                  Error
                </span>
              </div>
            </div>
          </div>

          <div class="material-form modal__material-form--email">
            <div class="material-form__input-text modal__input-text--email">
              <div class="group">
                <input
                  data-message="The format of your email is incorrect"
                  type="text"
                  placeholder=""
                  required=""
                  class="inpField"
                  v-model="email"
                />

                <span class="highlight"></span>
                <span class="bar"></span>

                <label>
                  Email
                </label>

                <span class="error-message">
                  Error
                </span>
              </div>
            </div>
          </div>

          <button type="button" class="button button--secondary modal__button--secondary">
            Cancel
          </button>

          <button type="button" class="button button--primary" @click="addUser">
            Add User
          </button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'AddModal',
  data() {
    return {
      firstName: '',
      lastName: '',
      email: '',
    };
  },
  methods: {
    buildUser(firstName, lastName, email) {
      return {
        first_name: firstName,
        last_name: lastName,
        email,
        creator_id: 1,
        created_date: new Date(),
      };
    },
    addUser() {
      const user = this.buildUser(this.firstName, this.lastName, this.email);
      this.$props.dispatschAddUser(user);
    },
  },
  props: ['dispatschAddUser'],
};
</script>
