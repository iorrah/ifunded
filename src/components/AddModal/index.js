// eslint-disable-next-line import/newline-after-import
import AddModal from './AddModal.vue';
export default AddModal;
