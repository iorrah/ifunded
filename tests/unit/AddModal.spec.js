import { mount } from '@vue/test-utils';
import AddModal from '../../src/components/AddModal/index';

const data = () => ({
  firstName: 'Jane',
  lastName: 'Doe',
  email: 'jane.doe#email.com',
  errorMessage: 'Invalid email address',
});

const propsData = {
  dispatschEditUser: jest.fn(),
  closeAddModal: jest.fn(),
};

const props = { data, propsData };

describe('AddModal', () => {
  test('renders correctly', () => {
    const wrapper = mount(AddModal, props);
    expect(wrapper.element).toMatchSnapshot();
  });
});
