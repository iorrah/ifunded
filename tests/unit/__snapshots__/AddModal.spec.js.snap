// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddModal renders correctly 1`] = `
<div
  class="modal__overlay"
  dispatschedituser="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
>
  <div
    class="modal__aligner"
  >
    <div
      class="modal__card"
    >
      <div
        class="modal__header"
      >
        <h2
          class="modal__title"
        >
          Add user
        </h2>
         
        <img
          class="modal__close-modal"
          src="../../assets/cross.svg"
        />
      </div>
       
      <div
        class="modal__body"
      >
        <div
          class="material-form"
        >
          <div
            class="material-form__input-text modal__input-text--first-name"
          >
            <div
              class="group"
            >
              <input
                class="inpField"
                data-message="The format of your email is incorrect"
                placeholder=""
                required="required"
                type="text"
              />
               
              <span
                class="highlight"
              />
               
              <span
                class="bar"
              />
               
              <label>
                
                First name
              
              </label>
               
              <span
                class="error-message"
              >
                
                Error with Jane
              
              </span>
            </div>
          </div>
           
          <div
            class="material-form__input-text modal__input-text--last-name"
          >
            <div
              class="group"
            >
              <input
                class="inpField"
                data-message="The format of your email is incorrect"
                placeholder=""
                required="required"
                type="text"
              />
               
              <span
                class="highlight"
              />
               
              <span
                class="bar"
              />
               
              <label>
                
                Last name
              
              </label>
               
              <span
                class="error-message"
              >
                
                Error with Doe
              
              </span>
            </div>
          </div>
        </div>
         
        <div
          class="material-form modal__material-form--email"
        >
          <div
            class="material-form__input-text modal__input-text--email"
          >
            <div
              class="group"
            >
              <input
                class="inpField"
                data-message="The format of your email is incorrect"
                placeholder=""
                required="required"
                type="text"
              />
               
              <span
                class="highlight"
              />
               
              <span
                class="bar"
              />
               
              <label>
                
                Email
              
              </label>
               
              <span
                class="error-message"
              >
                
                Error with jane.doe#email.com
              
              </span>
            </div>
          </div>
        </div>
         
        <p
          class="modal__error"
        >
          
          Invalid email address
        
        </p>
         
        <button
          class="button button--secondary modal__button--secondary"
          type="button"
        >
          
          Cancel
        
        </button>
         
        <button
          class="button button--primary"
          type="button"
        >
          
          Add User
        
        </button>
      </div>
    </div>
  </div>
</div>
`;
